source/index-management.rst

```diff
@@ -821,6 +821,11 @@ Common API Components
      * This option is only supported by servers >= 4.4.
      */
     hidden: Boolean;
+
+    /**
+     * Optionally specifies that this index is clustered.
+     */
+     clustered: Boolean;
```

Comment:
we might want to call out the server version where this is introduced similar to `hidden` above this.

also, is it possible to create a clustered index on an existing collection? the [docs](https://www.mongodb.com/docs/v6.0/core/clustered-collections/#limitations) seem to imply no:
> You cannot transform a non-clustered collection to a clustered collection, or the reverse.

if not, it's a little funny to have this here since drivers typically use this type for their `createIndexes` helpers... but on the flipside, we probably want it here for anyone who uses this type in their `listIndexes` output. 

we at least probably should call this out in the docstring because I think otherwise drivers may allow users to specify it here. in Swift anyway it would be easy enough to make it so users can only read but not set/mutate this property, but we can set it internally when decoding a `listIndexes` response. could we support something similar in Rust?

Reply:
Added version information.

The design doc actually [explicitly says](https://docs.google.com/document/d/1hRnqe1pq7ErcguJ5Bl8si1kHGKlZg0h25qsa3roASuo/edit#bookmark=id.oh101ver26rd) that `createIndexes` should accept `clustered`; I don't know how to reconcile that with the doc you linked.

Reply:
I got some clarification from this slack thread: https://mongodb.slack.com/archives/C2RCHGB2L/p1650576866908169 

given this, I suggest we note in the docstring here that this is not a valid option to provide to `createIndexes`, but can appear in the options returned for an index via `listIndexes`, and that if users want to create a clustered index, they need to create a new collection using the `clusteredIndex` option.

for the Rust impl it seems like we'll have to include it in `IndexOptions`, since that type models the response of to `createIndexes`, but can we do something to prevent users from actually setting the property themselves / omit the builder method for this property somehow?

Reply:
I didn't notice this earlier, but if this option is only relevant to `listIndexes` (and not `createIndexes` or any of the other helpers in this specification), should this be mentioned in the [Enumerating Indexes](https://github.com/mongodb/specifications/blob/master/source/enumerate-indexes.rst) specification? I suppose not, since that spec doesn't talk at all about modeling the response.

Reply:
it seems weird to me that we have two separate specs around indexes. why is enumerating indexes not just part of index management?
it feels like this could go either place, but since it seems like this option will mostly only matter for any drivers that use `IndexOptions` to model the result of `listIndexes` I lean toward keeping it here with the definition of the type